### PR TITLE
feat: first prototype version of the ts sdk

### PR DESCRIPTION
**Pull request by 0xterminator**
Items still under WIP:

- [ ] Unit tests
- [ ] More examples (2 already implemented)
- [ ] Annotations, function descriptions, README
- [ ] Unified serialization/deserialization between Rust and Ts. For now the payload shall remain as Uint8Array as we do not have the ability to deserialize it back using bincode or postcard (only available in Rust)

### CODE REVIEW

File: packages/fuel-streams/src/data/block/streamable.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import type { Block } from 'fuels';
+import { Streameable } from '../../streams/stream';
+import type { StreamData } from '../../streams/streamData';
+import { StreamEncoder } from '../../streams/streamEncoder';
+import { BlocksSubject } from './subjects';
+
+export class StreamedBlock extends Streameable<Block> {
```
Comment (Jurshsmith) [inline packages/fuel-streams/src/data/block/streamable.ts:7]: For consistency:

```suggestion
export class BlockStream extends Streameable<Block> {
```

File: packages/fuel-streams/src/data/logs/subjects.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    ).parseWildcard();
+  }
+
+  parse(): string {
+    const parts = [
+      'logs',
+      this.blockHeight?.toString() || '*',
+      this.txId || '*',
+      this.receiptIndex?.toString() || '*',
+      this.logId || '*',
```
Comment (Jurshsmith) [inline packages/fuel-streams/src/data/logs/subjects.ts:46]: These can be made dynamic like we have in Rust's, but this time at run-time

File: packages/fuel-streams/src/streams/stream.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    });
+    return new Stream<T>(store, client);
+  }
+
+  public getStore(): KV {
+    return this.store;
+  }
+
+  private prefixFilterSubject(subject: string): string {
+    return `$KV.*.${subject}`;
```
Comment (Jurshsmith) [inline packages/fuel-streams/src/streams/stream.ts:66]: This implementation is an hack based on the rust `asyn_nats`'s key-value store implementation. I imagine this being very peculiar to that library. Is it by any chance the same behavior with `@nats-io`?

**Review by Jurshsmith — CHANGES_REQUESTED** [review]

File: packages/fuel-streams/src/data/logs/subjects.ts (hunk 2 above)
Comment (0xterminator) [reply packages/fuel-streams/src/data/logs/subjects.ts:46]: @Jurshsmith can you elaborate more  ? I dont think I follow. ...

**Pull request merged**